src/apps/vpn/platforms/ios/iostunnelmanager.swift

```diff
@@ -84,12 +91,12 @@ class TunnelManager {
             if tunnel == nil {
                 Logger.global?.log(message: "Creating the tunnel")
                 self!.tunnel = .success(NETunnelProviderManager())
+                completionHandler();
```

Comment:
@mcleinman Not calling the handler to finalize initialization in all possible branches was what caused the VPN toggle not to work. Unit tests would have caught this. Do we have already a ticket to create Swift unit tests infra?

Reply:
I filed https://mozilla-hub.atlassian.net/browse/VPN-5048 to write the tests. I will block on the infra ticket, if it exists.

Reply:
Thanks!